tawkto/upgrades/version.070.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+if ( ! class_exists( 'TawkToUpgradeBase' ) ) {
+	require_once dirname( __FILE__ ) . '/base.php';
+}
+
+use Tawk\Helpers\PathHelper;
+use Tawk\Helpers\Common as CommonHelper;
+
+/**
+ * Upgrade for release version 0.7.0
+ */
+class TawkToUpgradeVersion070 extends TawkToUpgradeBase {
+	const VERSION = '0.7.0';
+
+	/**
+	 * Migration for url patterns with ending wildcards. (ex. https://www.example.com/path/to/somewhere/*)
+	 *
+	 * Adds the same url pattern without the ending wildcard (ex. https://www.example.com/path/to/somewhere)
+	 * to adjust with the new pattern matching lib.
+	 */
+	public static function upgrade() {
+		$visibility            = get_option( TawkTo_Settings::TAWK_VISIBILITY_OPTIONS );
+		$included_url_list     = array_map( 'trim', preg_split( '/,/', $visibility['included_url_list'] ) );
+		$excluded_url_list     = array_map( 'trim', preg_split( '/,/', $visibility['excluded_url_list'] ) );
+		$new_included_url_list = array();
+		$new_excluded_url_list = array();
+
+		foreach ( $included_url_list as $included_url ) {
+			if ( empty( $included_url ) ) {
+				continue;
+			}
+
+			$new_included_url_list[] = $included_url;
+
+			$wildcard = PathHelper::get_wildcard();
+
+			if ( false === CommonHelper::text_ends_with( $included_url, $wildcard ) ) {
+				continue;
+			}
+
+			$new_included_url = rtrim( $included_url, '/' . $wildcard );
+			if ( in_array( $new_included_url, $included_url_list, true ) ) {
+				continue;
+			}
+
+			$new_included_url_list[] = $new_included_url;
```

Comment:
List should be unique. We may have added a URL that already exists in list

Reply:
Added the fix on this PR Added the fix on this PR https://github.com/alvinjohnsonso/tawk-wordpress/pull/27/files